src/band/davidson.hpp

```diff
@@ -504,36 +692,76 @@ davidson(Hamiltonian_k<real_type<T>>& Hk__, int num_bands__, int num_mag_dims__,
+                            sphi->copy_from(*sphi_extra, num_extra_phi, 0, 0, 0, N + expand_with - num_extra_phi);
+                        } else {
+                            Hk__.apply_fv_h_o(false, false, N, expand_with, *phi, hphi.get(), sphi.get());
+                        }
+                        orthogonalize<T>(ctx.spla_context(), ctx.preferred_memory_t(), ctx.blas_linalg_t(),
+                                         spin_range(nc_mag ? 2 : 0), *phi, *hphi, *sphi, N, expand_with, H, *res, true);
+                    } else {
+                        /* for pseudopotential case we first project out the old subspace; this takes little less
+                         * operations and gives a slighly more stable procedure, especially for fp32 */
+                        project_out_subspace<T, F>(ctx.spla_context(), spin_range(nc_mag ? 2 : 0), *phi, *sphi, N, expand_with, H);
+                        Hk__.template apply_h_s<T>(spin_range(nc_mag ? 2 : ispin_step), N, expand_with, *phi, hphi.get(), sphi.get());
+                        orthogonalize<T>(ctx.spla_context(), ctx.preferred_memory_t(), ctx.blas_linalg_t(),
+                                         spin_range(nc_mag ? 2 : 0), *phi, *hphi, *sphi, N, expand_with, H, *res, false);
+                        if (extra_ortho__) {
+                            orthogonalize<T>(ctx.spla_context(), ctx.preferred_memory_t(), ctx.blas_linalg_t(),
+                                             spin_range(nc_mag ? 2 : 0), *phi, *hphi, *sphi, N, expand_with, H, *res, false);
+                        }
+                    }
+                    Band(ctx).set_subspace_mtrx<T, F>(N, expand_with, num_locked, *phi, *hphi, H, &H_old);
+                    break;
+                }
+                case davidson_evp_t::overlap: {
+                    project_out_subspace<T, F>(ctx.spla_context(), spin_range(nc_mag ? 2 : 0), *phi, *phi, N, expand_with, H);
+                    if (ctx.full_potential()) {
+                        Hk__.apply_fv_h_o(true, false, N, expand_with, *phi, nullptr, sphi.get());
+                    } else {
+                        Hk__.template apply_h_s<T>(spin_range(nc_mag ? 2 : ispin_step), N, expand_with, *phi, nullptr, sphi.get());
+                    }
+                    orthogonalize<T>(ctx.spla_context(), ctx.preferred_memory_t(), ctx.blas_linalg_t(),
+                                     spin_range(nc_mag ? 2 : 0), *phi, *sphi, N, expand_with, H, *res, false);
+                    if (extra_ortho__) {
+                        orthogonalize<T>(ctx.spla_context(), ctx.preferred_memory_t(), ctx.blas_linalg_t(),
+                                         spin_range(nc_mag ? 2 : 0), *phi, *sphi, N, expand_with, H, *res, false);
+                    }
+                    Band(ctx).set_subspace_mtrx<T, F>(N, expand_with, num_locked, *phi, *sphi, H, &H_old);
+                    break;
+                }
+            }
```

Comment:
i would put a default case here with an error although it might never be triggered. Just cosmetics though

Reply:
I think for enum types the compiler will issue a warning, that value is not handled in the switch. We are safe here.